frontends/web/src/api/lightning.ts

```diff
@@ -0,0 +1,961 @@
+  return await apiPost('lightning/config', data);
+};
+
+export const postActivateNode = async (): Promise<void> => {
+  return postApiResponse<void, undefined>('lightning/activate-node', undefined, 'Error calling postActivateNode');
+};
+
+export const postDeactivateNode = async (): Promise<void> => {
+  return postApiResponse<void, undefined>('lightning/deactivate-node', undefined, 'Error calling postDeactivateNode');
+};
+
+/**
+ * Breez SDK API interface
+ */
+
+export const getNodeInfo = async (): Promise<NodeState> => {
+  return getApiResponse<NodeState>('lightning/node-info', 'Error calling getNodeInfo');
+};
+
+export const getListPayments = async (params: ListPaymentsRequest): Promise<Payment[]> => {
+  return getApiResponse<Payment[]>(`lightning/list-payments?${qs.stringify(params, { skipNull: true })}`, 'Error calling getListPayments');
+};
+
+export const getOpenChannelFee = async (params: OpenChannelFeeRequest): Promise<OpenChannelFeeResponse> => {
+  return getApiResponse<OpenChannelFeeResponse>(
+    `lightning/open-channel-fee?${qs.stringify(params, { skipNull: true })}`,
+    'Error calling getOpenChannelFee'
+  );
+};
+
+export const getParseInput = async (params: ParseInputRequest): Promise<InputType> => {
+  return getApiResponse<InputType>(`lightning/parse-input?${qs.stringify(params, { skipNull: true })}`, 'Error calling getParseInput');
+};
+
+export const postSendPayment = async (data: SendPaymentRequest): Promise<SendPaymentResponse> => {
+  return postApiResponse<SendPaymentResponse, SendPaymentRequest>('lightning/send-payment', data, 'Error calling postSendPayment');
+};
+
+export const postReceivePayment = async (data: ReceivePaymentRequest): Promise<ReceivePaymentResponse> => {
+  return postApiResponse<ReceivePaymentResponse, ReceivePaymentRequest>(
+    'lightning/receive-payment',
+    data,
+    'Error calling postReceivePayment'
+  );
+};
+
+/**
+ * Subscriptions
+ */
+
+/**
+ * Returns a function that subscribes a callback on a "lightning/config"
+ * event to notify when a change to the lightning config has occurred.
+ * Meant to be used with `useSubscribe`.
+ */
+export const subscribeLightningConfig = (cb: TSubscriptionCallback<LightningConfig>) => {
+  return subscribeEndpoint('lightning/config', cb);
+};
+
+/**
+ * Returns a function that subscribes a callback on a "lightning/list-payments"
+ * event to notify when a change to the payments list has occurred.
+ * Meant to be used with `useSubscribe`.
+ */
+export const subscribeListPayments = (cb: TSubscriptionCallback<Payment[]>) => {
+  return subscribeEndpoint('lightning/list-payments', cb);
+};
+
+/**
+ * Returns a function that subscribes a callback on a "lightning/node-state"
+ * event to receive the latest state of the lightning node.
+ * Meant to be used with `useSubscribe`.
+ */
+export const subscribeNodeState = (cb: TSubscriptionCallback<NodeState>) => {
+  return subscribeEndpoint('lightning/node-info', cb);
+};
```

Comment:
This file is crazy. Was this generated somehow, if yes how?

Also there seems to be un-used stuff, i.e. EnvironmentType or ReverseSwapFeesRequest

Reply:
This code is generated from rust for our react native package.  Currently it was just copy/pastes. Let me know if you want me to remove any unused types.

Reply:
It is at the moment unlcear how to deal with this file.

We could also remove the unused types / api calls later on staging. 

But if this is a generated file, we probably shouldn't do manual changes?

are you planing to re-generate this again if the rust package changes?